llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -1147,27 +1103,29 @@ void WaitcntBrackets::print(raw_ostream &OS) const {
       // Print vgpr scores.
       unsigned LB = getScoreLB(T);
 
-      for (int J = 0; J <= VgprUB; J++) {
-        unsigned RegScore = getRegScore(J, T);
+      for (auto &[ID, Info] : VMem) {
+        unsigned RegScore = Info.Scores[T];
         if (RegScore <= LB)
           continue;
         unsigned RelScore = RegScore - LB - 1;
-        if (J < FIRST_LDS_VGPR) {
-          OS << RelScore << ":v" << J << " ";
-        } else {
-          OS << RelScore << ":ds ";
+        if (ID < REGUNITS_END) {
+          OS << RelScore << ":vRU" << ID << " ";
+        } else if (ID >= LDSDMA_BEGIN && ID < LDSDMA_END) {
```

Comment:
This could be an assert, right?

```suggestion
        } else {
          assert(ID >= LDSDMA_BEGIN && ID < LDSDMA_END);
```